### PR TITLE
Generate a config if one does not exist

### PR DESCRIPTION
**Pull request by ajaska**
I've never written go before

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -15,15 +18,46 @@ type Config struct {
 	Slack slack
 }
 
-// ParseConfig reads a ./config.toml file and retruns a parsed config
-func ParseConfig() (*Config, error) {
+// parses a config if it exists, or generates a new one
```
Comment (nolanlum) [inline config.go]: apparently the linter complains if you don't start these doc comments with the method name...so it has to read `// LoadConfig ...`

File: token/token.go [hunk 2]
```diff
@@ -10,70 +11,69 @@ import "github.com/nlopes/slack"
...
-    fmt.Println(loginResponse.Token)
+func GetSlackToken() (string, error) {
+	var domain, email, token string
+
+	fmt.Print("Team domain (*.slack.com): ")
+	fmt.Scanln(&domain)
+
+	resp, err := http.PostForm("https://slack.com/api/auth.findTeam", url.Values{"domain": {domain}})
+	if err != nil {
+		return token, err
```
Comment (nolanlum) [inline token/token.go]: `return nil, err` here for the error cases my dude

Comment (nolanlum) [reply token/token.go]: err, `return "", err` is what I meant...

File: token/token.go [hunk 3]
```diff
@@ -10,70 +11,69 @@ import "github.com/nlopes/slack"
...
+	}
+
+	var loginResponse loginResponseFull
+	err = json.Unmarshal(body, &loginResponse)
+	if err != nil {
+		return token, err
+	}
+
+	token = loginResponse.Token
+	return token, nil
```
Comment (nolanlum) [inline token/token.go]: seems fine to just `return loginResponse.Token, nil`

File: token/token.go (hunk 2 above)
Comment (ajaska) [reply token/token.go]: o

File: config.go [hunk 4]
```diff
@@ -15,15 +18,46 @@ type Config struct {
...
+	}
+	fmt.Println("Done")
+	return &conf, nil
+}
 
+// parseConfig reads a toml string and returns a parsed config
+func parseConfig(tomlData string) (*Config, error) {
 	var conf Config
-	if _, err = toml.Decode(string(tomlData), &conf); err != nil {
+	if _, err := toml.Decode(string(tomlData), &conf); err != nil {
```
Comment (nolanlum) [inline config.go]: can remove the `string()` here for shiggles

**Pull request merged**